docs/features/readonly-variables-and-parameters.md

```diff
@@ -0,0 +1,61 @@
+### High Level Language Proposal:  
+Allow 'readonly' to be used on local variables and on parameters. 
+
+'readonly' on local variables and 'parameters' will work similarly as it does for 'fields' (with important differences listed in the next paragraph).  Namely, 'readonly' local variables and parameters cannot be written to in any way (i.e. through assignment, use of ```++/+=```, passing as a 'ref/out', etc.). 
+
+Important differences:
+unlike 'readonly' fields, 'readonly' locals must have an initializer.
+unlike 'readonly' fields (which can be written to multiple times in constructors), 'readonly' locals can only have the initializer and can never be written to again.
```

Comment:
Agreed.  I also think the label case is pretty rare in code in practice.  The 'switch' case is something i would prefer to not disallow.  I think a very simple way to disallow in switches would be to just disallow if the switch has any goto's in it.  In that case (har har), then we wouldn't allow it.  But for nearly all the switches out there that don't use 'goto', you'd be allowed to have readonly locals.